test/byContract/IexecPoco/04_finalize.test.ts

```diff
@@ -0,0 +1,331 @@
+            for (const worker of winningWorkers) {
+                expect(await iexecPoco.balanceOf(worker.address)).to.be.equal(
+                    workerTaskStake + workerReward,
+                );
+                expect(await iexecPoco.frozenOf(worker.address)).to.be.equal(0);
+                expect(await iexecPoco.viewScore(worker.address)).to.be.equal(1);
+            }
+            expect(await iexecPoco.balanceOf(losingWorker.address)).to.be.equal(0);
+            expect(await iexecPoco.frozenOf(losingWorker.address)).to.be.equal(0);
+            // TODO: Add score history to losing worker to test score punishing
+            expect(await iexecPoco.viewScore(losingWorker.address)).to.be.equal(0);
+            // TODO: Update test with non-empty kitty
+            expect(await iexecPoco.balanceOf(kittyAddress)).to.be.equal(0);
+            expect(await iexecPoco.frozenOf(kittyAddress)).to.be.equal(0);
+        });
+
+        it('Should finalize task of deal payed by requester', async function () {
+            const { orders } = buildOrders({
+                assets: ordersAssets,
+                requester: requester.address,
+                prices: ordersPrices,
+            });
+            const { dealId, taskId, taskIndex, dealPrice } =
+                await iexecWrapper.signAndMatchOrders(orders);
+            await iexecPocoAsAnyone.initialize(dealId, taskIndex).then((tx) => tx.wait());
+            const workerTaskStake = await iexecPoco
+                .viewDeal(dealId)
+                .then((deal) => deal.workerStake.toNumber());
+            const { resultHash, resultSeal } = buildResultHashAndResultSeal(
+                taskId,
+                resultDigest,
+                worker1,
+            );
+            const schedulerSignature = await buildAndSignContributionAuthorizationMessage(
+                worker1.address,
+                taskId,
+                enclaveAddress,
+                scheduler,
+            );
+            await iexecWrapper.depositInIexecAccount(worker1, workerTaskStake);
+            await iexecPoco
+                .connect(worker1)
+                .contribute(
+                    taskId,
+                    resultHash,
+                    resultSeal,
+                    enclaveAddress,
+                    '0x',
+                    schedulerSignature,
+                )
+                .then((tx) => tx.wait());
+            await iexecPoco
+                .connect(worker1)
+                .reveal(taskId, resultDigest)
+                .then((tx) => tx.wait());
+            expect(await iexecPoco.balanceOf(requester.address)).to.be.equal(0);
+            expect(await iexecPoco.frozenOf(requester.address)).to.be.equal(taskPrice);
+            expect(await iexecPoco.balanceOf(sponsor.address)).to.be.equal(0);
+            expect(await iexecPoco.frozenOf(sponsor.address)).to.be.equal(0);
+
+            await expect(iexecPoco.connect(scheduler).finalize(taskId, results, '0x'))
```

Comment:
Fix `0x` to `const enclaveSign` ?

Reply:
Yes, see https://github.com/iExecBlockchainComputing/PoCo/pull/79/commits/d79087c4ce3966599b6e49ac8ba2e5603200f975! The one pointed out is not updated, since not related to enclave, nor used somewhere else :)